### ISSUE TITLE
`Configurables`| `LoadConfiguration` misleading naming wrt ownership

### ISSUE THREAD

**Issue by segfault-magnet** [FuelLabs/fuels-rs#1089]
Configurables have methods for each configurable that start with `set_`. 

They also unexpectedly consume `self` which is unexpected coming from a setter.

Maybe we should rename this to `with_xyz` instead.

We can also still generate the `set` methods but make them deprecated with a note pointing to the `with_xyz` new alternatives.

The same is present on `LoadConfiguration` where `set_` methods take ownership of the object.

Also it seems the ContractCallHandler does this. We might be having a setter crisis :)

**Comment by segfault-magnet** [comment]
Also it seems that `ScriptCallHandler` is also confused about `set` vs `with`

**Comment by MujkicA** [comment]
After giving this a go I realised that changing all builder type methods to use 'with_' instead of 'set_' is a considerable commitment.
Here is the draft PR for reference #1096. Note It still doesn't include the renaming of the transaction builder setter methods (set inputs, outputs, witnesses etc.).

@FuelLabs/sdk-rust what's the opinion, should we proceed with this? Consistency would be nice, especially if we consider cases like tx builder and tx where one has setter and the other has 'with_' methods. The downside is that the users will have to update sooner or later and we need to keep track of the deprecated code until then.
